lib/layout.js

```diff
@@ -17,6 +17,127 @@ var _ = require("./lodash"),
 
 module.exports = layout;
 
+function hasMultiplePorts(g, node) {
+  var inPorts = {};
+  var outPorts = {};
+
+  _.forEach(g.nodeEdges(node), function (edge) {
+    if (edge.v === node && edge.vport) {
+      inPorts[edge.vport] = {};
```

Comment:
I am just setting this to any value so a property gets added the `inPorts` object because the number of properties in the object will be counted later.  I could have used an array, but then I would have had to check if the value was already in the array before adding so I went this route. Perhaps I could have used a boolean instead of an empty object - that might have been more clear.